x/evm/keeper/state_transition.go

```diff
@@ -560,36 +569,20 @@ func (k *Keeper) prepareTxForSgx(ctx sdk.Context, msg core.Message, cfg *EVMConf
 			DebugTrace:      cfg.DebugTrace,
 			NoBaseFee:       cfg.FeeMarketParams.NoBaseFee,
 			EvmDenom:        cfg.Params.EvmDenom,
-			Overrides:       ToGOB64(cfg.Overrides),
+			Overrides:       overrides,
 		},
 	}
 
-	return sgxRPCClient.PrepareTx(args, &PrepareTxReply{})
-}
-
-func (k *Keeper) runRPCServer(ctx sdk.Context, msg core.Message, cfg *EVMConfig) error {
-	defer func() {
-		if r := recover(); r != nil {
-			ctx.Logger().Debug("recovered from panic", "error", r)
-		}
-	}()
-
-	// TODO Think about whether the RPC server should be persistent or ephemeral
-	//  - If it's persistent, we need to handle the lifecycle of the RPC server
-	//  - If it's ephemeral, we need to create a new RPC server for each message
-	//  The current implementation is ephemeral.
-	srv := &EthmRpcServer{k: k, ctx: ctx, msg: msg, evmCfg: cfg}
-	rpc.Register(srv)
-	rpc.HandleHTTP()
-
-	// TODO handle port customization
-	l, err := net.Listen("tcp", ":9093")
+	reply := &PrepareTxReply{}
+	err = sgxRPCClient.PrepareTx(args, reply)
 	if err != nil {
-		// TODO handle error
-		panic(err)
+		return err, 0
 	}
-	// TODO Handle shutdown
-	go http.Serve(l, nil)
 
-	return nil
+	// Store handler id
+	handlerId := reply.HandlerId
+	// Snapshot the ctx
+	k.preparedCtxs[handlerId] = &ctx
```

Comment:
@kenta-mori3322 we should also delete the entry from the map once it's not used anymore. This should probably happen at the end of the MsgServer handler and Query handler.